src/evaluators/UnaryExpression.js

```diff
@@ -95,43 +94,29 @@ function evaluateDeleteOperation(expr: Value | Reference, realm: Realm) {
   return new BooleanValue(realm, bindings.DeleteBinding(referencedName));
 }
 
-function generateRuntimeCall(ref: Reference | Value, ast: BabelNodeUnaryExpression, strictCode: boolean, realm: Realm) {
-  invariant(ref instanceof Reference);
-  let baseValue = Environment.GetBase(realm, ref);
-  invariant(baseValue instanceof Value);
-  let propertyName = Environment.GetReferencedName(realm, ref);
-  invariant(typeof propertyName === "string");
-
-  if (!baseValue.isSimpleObject()) {
-    Leak.leakValue(realm, baseValue, ast.loc);
-  }
-  return AbstractValue.createTemporalFromBuildFunction(realm, Value, [baseValue], nodes => {
-    let arg;
-    if (typeof propertyName === "string") {
-      arg = t.isValidIdentifier(propertyName)
-        ? t.memberExpression(nodes[0], t.identifier(propertyName), false)
-        : t.memberExpression(nodes[0], t.stringLiteral(propertyName), true);
-    } else {
-      arg = t.memberExpression(nodes[0], nodes[1], true);
-    }
-    return t.unaryExpression(ast.operator, arg, ast.prefix);
-  });
+function generateRuntimeCall(expr: Value | Reference, ast: BabelNodeUnaryExpression, realm: Realm) {
+  let value = Environment.GetValue(realm, expr);
+  invariant(value instanceof AbstractValue);
```

Comment:
Why do you believe this must be an AbstractValue?

Reply:
`createFromUnaryOp` expects an `AbstractValue` as its third argument.

Reply:
Why do you believe that GetValue will always produce an abstract value. I can easily construct a call to generateRuntimeCall where that will not be the case, so you are asserting a precondition for the expr parameter. You need to make that explicit in a comment and you need to demonstrate that every call site satisfies the precondition.